metagraph/src/graph/representation/succinct/dbg_succinct.hpp

```diff
@@ -183,6 +196,8 @@ class DBGSuccinct : public DeBruijnGraph {
 
     const mtg::kmer::KmerBloomFilter<>* get_bloom_filter() const { return bloom_filter_.get(); }
 
+    std::shared_ptr<CachedView> get_cached_view(size_t cache_size = 100'000) const;
```

Comment:
Better return `CachedView`. You can always create a shared_ptr from it (there should be a move constructor) if you need it.

Reply:
`CachedView` is a pure abstract class, so this method can't return the actual object.

Reply:
Then return unique_ptr. You can convert it to shared when needed. Otherwise, it will be more clear that this thing is a new object.